draft-ietf-tls-tlsflags.xml

```diff
@@ -131,13 +131,20 @@
             <t> A server that supports this extension and also supports at least one of the flag-type features
                 that use this extension and that were declared by the ClientHello extension SHALL send this
                 extension with the intersection of the flags it supports with the flags declared by the client.
-                The intersection operation MAY be implemented as a bitwise AND. The server may need to send two
-                tls_flags extensions, one in the ServerHello and the other in the EncryptedExtensions message.
-                It is up to the document for the specific feature to determine whether support should be
-                acknowledged in the ServerHello or the EncryptedExtensions message.</t>
+                The intersection operation may be implemented as a bitwise AND. The server may need to send multiple
+                tls_flags extensions, up to one in each of the response messages that may carry extensions: 
+                ServerHello (SH), EncryptedExtensions (EE), Certificate (CT), and HelloRetryRequest (HRR). 
+                It is up to the document for the specific feature to determine which of these should be used to
+                acknowledge support.</t>
             <t> A server MUST NOT indicate support for any flag-type feature not previously indicated by the 
-                client. It MUST NOT include this extension in either message (ServerHello or EncryptedExtensions)
+                client in the ClientHello in any response message. It MUST NOT include this extension in any of the four messages 
                 if it has no appropriate flag-type to indicate. This extension MUST NOT be included empty.</t>
+            <t> A server MAY indicate support for flag-type features in a flags extension of CertificateRequest (CR). 
+                Flag-type extensions that have been indicated in CR (and only such extensions) MAY be also be present
+                in a flags extension of the Certificate (CT) message sent by the client.</t>
```

Comment:
This probably needs to have a version of the MUST NOT from line 139.  Something like.

> A client MUST NOT indicate support for any flag-type feature in a Certificate message that was not previously indicated by the server in its CertificateRequest message.

Reply:
👍